src/ast_utils.h

```diff
@@ -187,7 +189,14 @@ struct EffectAnalyzer : public PostWalker<EffectAnalyzer> {
   }
 
   void visitCall(Call *curr) { calls = true; }
-  void visitCallImport(CallImport *curr) { calls = true; }
+  void visitCallImport(CallImport *curr) {
+    calls = true;
+    if (debugInfo) {
+      // debugInfo call imports must be preserved very strongly, do not
```

Comment:
To do that properly, we should also refactor the name of those intrinsics, which I notice now is `emscripten_debuginfo`, but should probably be moved out of `asm2wasm.h` into `debugInfo.h` or such and renamed to `binaryen_debuginfo` (it's in `asm2wasm.h` now because I prototyped the approach there). Maybe worth doing later.